Sources/Purchasing/OfferingsManager.swift

```diff
@@ -289,6 +301,32 @@ extension OfferingsManager {
 
     }
 
+    /// Determines the behavior when fetching `Offerings`
+    internal enum FetchBehavior {
+
+        /// Returns values from the cache, or throws an error if not available.
+        case cachedOrFetched
+
+        /// Always fetch the most up-to-date data.
+        case fetchCurrent(reason: FetchReason)
+
+        static let `default`: Self = .cachedOrFetched
+
+        var fetchReason: String? {
+            switch self {
+            case .cachedOrFetched:
+                return nil
+            case .fetchCurrent(reason: let reason):
+                return reason.rawValue
+            }
+        }
+    }
+
+    internal enum FetchReason: String {
+        case manualSyncCustomAttributes = "custom_attributes_sync_manual"
+        case automaticSyncCustomAttributes = "custom_attributes_sync_automatic"
+    }
```

Comment:
Replaced our boolean with a enum for fetching behavior. Don't know if I love the name `FetchBehavior` (it differs from our customer one) but it felt different enough that our existing `FetchPolicy` in this same file.

`fetchCurrent` takes a `FetchReason` because we want to know why we are going straight for new offerings